### PR TITLE
Search bar for settings

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Close https://github.com/enso-org/cloud-v2/issues/1198
  - Add search bar for settings
    - Note that, *unlike* the asset search bar, it has neither suggestions nor tags (name:, description:) - I don't think those add any value here (and definitely would complicate the implementation by a lot)
  - Convert settings objects to JS so that they can be filtered by search

UX changes required for this PR:
- Switched `MembersSettingsSection` to use `react-query` to avoid loading the members list every time a user types into the search bar

Incidental changes added to make this PR simpler:
- Refactor search bar to be provided by a React Context so that state no longer needs to be hoisted all the way up into `Dashboard` and back down to `TopBar`

### Important Notes
- While I think highlighting the tab of the currently visible section would be a nice UX upgrade, I think it might be a bit complex for at least this initial PR.
  - It's definitely possible, but almost certainly requires Intersection Observers, plus a new constant to map each Settings Section ID back to its parent Settings Tab ID

### Screencast

Note: The end of this recording contains a bug where the "user account" section was not being matched. This bug has since been fixed.

https://github.com/enso-org/enso/assets/4046547/1023dbd6-907d-4add-a41a-d4b59e46f9b2

### Code Review tips
- `settingsData.ts` contains the settings converted to JS objects
- `Settings.tsx` contains the new filtering logic
- `Settings/Settings{Tab,Section,Entry,InputEntry,CustomEntry}` contains the components responsible for rendering the new JS objects
  - The old components which manually constructed the tabs from their constituent components have been removed

### Testing instructions
- Mostly testing that the search works as expected
- Partly testing to make sure nothing was broken by the refactors - either CSS wise or otherwise.
  - Most things *should* retain 100% of their original logic - search for `SettingsEntryType.custom` in `settingsData`

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] ~~Unit tests have been written where possible.~~

### CODE REVIEW

File: app/ide-desktop/lib/dashboard/src/layouts/SearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          <FocusRing placement="before">
+            <aria.SearchField
+              aria-label={label}
+              className="relative grow before:text before:absolute before:inset-x-button-focus-ring-inset before:my-auto before:rounded-full before:transition-all"
+              value={query}
+              onKeyDown={event => {
+                event.continuePropagation()
+              }}
+            >
+              <aria.Input
```
Comment (MrFlashAccount) [inline app/ide-desktop/lib/dashboard/src/layouts/SearchBar.tsx:51]: Why did you put input in searchfield ?

Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/layouts/SearchBar.tsx:51]: how else are you supposed to do it?

Comment (MrFlashAccount) [reply app/ide-desktop/lib/dashboard/src/layouts/SearchBar.tsx:51]: @somebody1234 according to the [docs](https://react-spectrum.adobe.com/react-spectrum/SearchField.html#read-only), `SearchField`'s already an input

Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/layouts/SearchBar.tsx:51]: that's react-spectrum - we use RAC (unstyled), not spectrum (styled)

Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/layouts/SearchBar.tsx:51]: see react-aria docs for comparison:
<https://react-spectrum.adobe.com/react-aria/SearchField.html#searchfield>

Comment (MrFlashAccount) [reply app/ide-desktop/lib/dashboard/src/layouts/SearchBar.tsx:51]: Oh yeah... sorry :)

**Comment by PabloBuchu** [comment]
Can't open Enso.. probably needs a merge with develop?

Screenshot 2024-06-10 at 13 55 40

**Comment by somebody1234** [comment]
yeah, there was an issue fixed on the weekend that breaks all currently open PRs

**Comment by somebody1234** [comment]
@PabloBuchu merged with develop

**Comment by PabloBuchu** [comment]
Ugh... this isn't working very well... As you can see on your video when I start typing it creates a mess on user screen.. also when I search for `picture` I see 2 partial components for profile and organization picture.. I dont think it should work that way..

Screenshot 2024-06-11 at 12 57 22

So the search should filter list of pages on the left (it can still search through the page content but avoid mixing their content) . Like in pycharm (see the video)


https://github.com/enso-org/enso/assets/2855109/8cbd6d81-5471-43a8-957a-891733e3442c

**Comment by somebody1234** [comment]
@PabloBuchu should be fixed now.

one notable difference of the *current* impl is that the current page's content is still *always* filtered - so a lot of the time there may be a blank page if there are no matches on the current page.

should be easy to change either way, so this is mostly just a heads up on how the current behavior works (as of latest commit)

**Comment by PabloBuchu** [comment]
Sorry can't build 

Screenshot 2024-06-18 at 14 26 07

**Comment by PabloBuchu** [comment]
1. Search bar is now 100%

Screenshot 2024-06-18 at 14 37 10

2. Also dashboard search changed position?
Screenshot 2024-06-18 at 14 37 59

3. Regarding how it works. How hard would it be to auto select always first page from the search results?

**Comment by somebody1234** [comment]
1. noted, fixed locally (along with other minor style fixes), pushing now
2. yes - this PR previously included a refactor of the search bar so that React state didn't have to get hoisted *all the way* up to the `Dashboard` component - however it was incompatible with the new design whih moves the asset search bar there anyway. so i figured the best solution is to put it where it will be soon anyway (see dashboard-tabs PR for the full new design, including the placement of the assets search bar)
  - basically the idea is, the code was already refactored away from the old style even before the new design was known - so it's probably less effort to refactor to the new design (simpler in this case) than to revert just to have to un-revert it again in the `dashboard-tabs` PR (not to mention risking undoing moving the search bar if this one ends up landing later than the `dashboard-tabs` PR)
3. should be doable enough - i'll take a look at it now

**Comment by somebody1234** [comment]
i guess one thing to note is that this doesn't yet have support for paywalls - it will likely need dedicated support since paywalls *replace* the entire page, no matter what the search results would normally be

**Comment by somebody1234** [comment]
ok, 3. should be implemented now as well

**Comment by somebody1234** [comment]
Should be merged after:

- #10203

which introduces paywalls to some settings pages which will need to be implemented in the new structure under which settings pages are stored

**Comment by somebody1234** [comment]
*almost* ready, just having issues with input validation making E2E tests fail

**Comment by PabloBuchu** [comment]
@somebody1234 can you merge develop tomorrow so I can reqa?

**Comment by PabloBuchu** [comment]
Small tweak. When I type `Acti` and I properly see `Activity log` page the first entry on left menu should be highlighted as selected. Currently selection is only aplied either on first entry `Account` ot last clicked by user. 

Screenshot 2024-06-26 at 11 54 24

**Comment by somebody1234** [comment]
@PabloBuchu should be implemented now. one other change i've made to make this fit in more, is if the user presses mouse/pointer, or moves keyboard focus into the element, the tab is switched permanently (otherwise it is changed back to the original tab when the query is changed)

**Comment by PabloBuchu** [comment]
QA ✅

**Review by PabloBuchu — APPROVED** [review]
